### PR TITLE
Buy album

### PR DESCRIPTION
**Pull request by 0xAaCE**
## Summary

Add buy album mutation

## Trello Card

https://trello.com/c/W8u9nJNx/15-comprar-albums

### CODE REVIEW

File: app/helper/token.js [hunk 1]
```diff
@@ -1,5 +1,20 @@
...
+  } catch (e) {
+    throw errors.defaultError(e.message);
+  }
+};
+
+exports.decode = token => {
+  try {
+    return jwt.verify(token, secretKey);
+  } catch (e) {
+    return null;
```
Comment (gonziesc) [inline app/helper/token.js]: why null?

File: app/models/trasaction.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+module.exports = (sequelize, DataTypes) => {
+  const Transaction = sequelize.define(
```
Comment (gonziesc) [inline app/models/trasaction.js]: do you think trasaction is an expressive name for this model?

File: app/services/album.js [hunk 3]
```diff
@@ -53,3 +55,33 @@ exports.getAllAlbums = (offset, limit, filter, orderBy) => {
       return processedAlbums.slice(offset, limit);
     });
 };
+
+exports.buyAlbum = (albumId, context) => {
```
Comment (gonziesc) [inline app/services/album.js:59]: good error managment!

File: test/server.spec.js [hunk 4]
```diff
@@ -2,16 +2,24 @@ const { createTestClient } = require('apollo-server-testing'),
...
-    schema,
-    formatError: err => ({
-      message: err.message,
-      statusCode: err.extensions.code
-    })
+ApolloServer.prototype.setContext = function setContext(newContext) {
+  this.context = newContext;
+};
+
+ApolloServer.prototype.resetContext = function resetContext() {
```
Comment (gonziesc) [inline test/server.spec.js:9]: do we really need this?

**Review by gonziesc — CHANGES_REQUESTED** [review]

File: test/server.spec.js (hunk 4 above)
Comment (0xAaCE) [reply test/server.spec.js:9]: It is necessary to imitate the function of context on the real server

File: app/models/trasaction.js (hunk 2 above)
Comment (0xAaCE) [reply app/models/trasaction.js]: Is BoughtAlbums better?

Comment (gonziesc) [reply app/models/trasaction.js]: thats better! also purchases would be a good name

File: app/helper/token.js (hunk 1 above)
Comment (gonziesc) [inline app/helper/token.js]: why you do nothing with the exception and return null?

**Review by gonziesc — APPROVED** [review]

**Pull request merged**